contracts/modules/capital/Pool.sol

```diff
@@ -122,73 +139,52 @@ contract Pool is IPool, MasterAwareV2, ReentrancyGuard {
   /* ========== ASSET RELATED VIEW FUNCTIONS ========== */
 
   function getAssetValueInEth(address assetAddress) internal view returns (uint) {
-    IERC20 token = IERC20(assetAddress);
 
     uint assetBalance;
-    if (assetAddress.code.length != 0){
-      try token.balanceOf(address(this)) returns (uint balance) {
+
+    if (assetAddress.code.length != 0) {
+      try IERC20(assetAddress).balanceOf(address(this)) returns (uint balance) {
         assetBalance = balance;
-      } catch{
+      } catch {
         // If balanceOf reverts consider it 0
       }
     }
 
     // If the assetBalance is 0 skip the oracle call to save gas
     if (assetBalance == 0) {
-      return 0; // ETH
+      return 0;
     }
 
     return priceFeedOracle.getEthForAsset(assetAddress, assetBalance);
   }
 
-  /**
-   * @dev Calculates total value of all pool assets in ether
-   */
+  ///
+  /// @dev Calculates total value of all pool assets in ether
+  ///
   function getPoolValueInEth() public override view returns (uint) {
 
     uint total = address(this).balance + swapValue;
-    uint investmentAssetsCount = investmentAssets.length;
-    uint coverAssetsCount = coverAssets.length;
-
-    for (uint i = 0; i < investmentAssetsCount; i++) {
-      uint assetValue = getAssetValueInEth(investmentAssets[i].assetAddress);
-      total = total + assetValue;
-    }
-
-    uint deprecatedCoverAssets = deprecatedCoverAssetsBitmap;
+    uint assetCount = assets.length;
 
     // Skip ETH (index 0)
-    for (uint i = 1; i < coverAssetsCount; i++) {
-      // Skip deprecated assets by looking at the bits that are on in deprecatedCoverAssetsBitmap
-      if ((1 << i) & deprecatedCoverAssets != 0) {
+    for (uint i = 1; i < assetCount; i++) {
+
+      if (assets[i].isAbandoned) {
```

Comment:
what happens to an asset that we flip to isAbandoned if it's still present in the pool?

Is the expected state we always have balance 0 for that asset before flipping this boolean to true?

Reply:
For deprecation, flipping `isCoverAsset` to false is sufficient. Abandoning should be done if the pool was blacklisted and we cannot transfer or maybe even use `balanceOf`, hence **not** requiring the balance to be zero is important.

Reply:
this clarifies it.


A cover asset essentially becomes an "investment" asset when deprecated.

it will still be moved around, but not allowed for covers.

then to leave it behind, we flip isAbandoned.

👍